GKE: Kubectl client Simple pod should support inline execution and attach

Kubectl client Simple pod should support inline execution and attach

https://k8s-testgrid.appspot.com/release-1.4-blocking#gke-1.2-1.4-upgrade-cluster

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gke-1.2-1.4-upgrade-cluster/6162

when I tried to debug this on my local, just got this error:
```
[root@localhost kubernetes]# go run hack/e2e.go -v -test --test_args='--ginkgo.focus=Kubectl\sclient\sSimple\spod\sshould\ssupport\sinline\sexecution\sand\sattach$'
2016/09/15 11:30:50 e2e.go:440: Running: get status
Local doesn't need special preparations for e2e tests
Client Version: version.Info{Major:"1", Minor:"3+", GitVersion:"v1.3.5-dirty", GitCommit:"b0deb2eb8f4037421077f77cb163dbb4c0a2a9f5", GitTreeState:"dirty", BuildDate:"2016-09-15T15:16:21Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
error: server version (&version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-beta.0-dirty", GitCommit:"44368db9916cc345ebef8b6fbde3cdf0dc9d79dc", GitTreeState:"dirty", BuildDate:"2016-09-15T13:54:48Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}) differs from client version (version.Info{Major:"1", Minor:"3+", GitVersion:"v1.3.5-dirty", GitCommit:"b0deb2eb8f4037421077f77cb163dbb4c0a2a9f5", GitTreeState:"dirty", BuildDate:"2016-09-15T15:16:21Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"})!
2016/09/15 11:30:50 e2e.go:446: Error running get status: exit status 1
2016/09/15 11:30:50 e2e.go:442: Step 'get status' finished in 46.285881ms
2016/09/15 11:30:50 e2e.go:418: Testing requested, but e2e cluster not up!
exit status 1
```

I set up a one node cluster with kubernetes 1.4 on my local virtual machine, and then I tried to run the e2e tests with kubernetes 1.3.5, and just got this error. Quite unlucky. :(

--- a/test/e2e/kubectl.go
+++ b/test/e2e/kubectl.go
@@ -459,6 +459,8 @@ var _ = Describe("Kubectl client", func() {
 
 		It("should support inline execution and attach", func() {
 			SkipUnlessServerVersionGTE(jobsVersion, c)
+			// if the version is >= 1.3.0 the job with --restart=Never is a pod, see: https://github.com/kubernetes/kubernetes/commit/d76fa8a1197a6329da3125c8bfb2202dfead1273
+			SkipUnlessServerVersionLTE(version.MustParse("v1.3.0-alpha.5"), c)
 
 			nsFlag := fmt.Sprintf("--namespace=%v", ns)
 

--- a/test/e2e/util.go
+++ b/test/e2e/util.go
@@ -376,6 +376,16 @@ func SkipUnlessServerVersionGTE(v semver.Version, c discovery.ServerVersionInter
 	}
 }
 
+func SkipUnlessServerVersionLTE(v semver.Version, c discovery.ServerVersionInterface) {
+	gte, err := serverVersionGTE(v, c)
+	if err != nil {
+		Failf("Failed to get server version: %v", err)
+	}
+	if gte {
+		Skipf("Not supported for server versions after %q", v)
+	}
+}
+
 // providersWithSSH are those providers where each node is accessible with SSH
 var providersWithSSH = []string{"gce", "gke", "aws"}
 

